### ISSUE TITLE
`Unexpected label` when argument is not used.

### ISSUE THREAD

**Issue by sporto** [gleam-lang/gleam#1098]
Consider a function like

```
fn greet(name name: String, title title: String) {
   name
}

greet(name: "Sam", title: "Mr")
```

Title is not used in the greet function, so the compiler suggest to use `_title`.

```
fn greet(name name: String, title _title: String) {
   name
}
```

However after doing that the compiler fails with:

```
           title: "Mr",
    │             ^^^^^^^^^ Unexpected label

You have already supplied all the labelled arguments that this constructor accepts.
```

I think that the caller should not concern itself with the fact that `title` is not used.

**Comment by lpil** [comment]
Ah yes, this does seem like a bug. Thanks for the report.

**Comment by shunjilin** [comment]
I believe this may be closed by #1111  (sorry, forgot to link the PR)

**Comment by lpil** [comment]
Thanks again!
